jiant/proj/main/modeling/taskmodels.py

```diff
@@ -276,14 +282,94 @@ def get_output_from_encoder(encoder, input_ids, segment_ids, input_mask) -> Enco
         EncoderOutput containing pooled and unpooled model outputs as well as any other outputs.
 
     """
+    model_arch = ModelArchitectures.from_encoder(encoder)
+    if model_arch in [
+        ModelArchitectures.BERT,
+        ModelArchitectures.ROBERTA,
+        ModelArchitectures.ALBERT,
+        ModelArchitectures.XLM_ROBERTA,
+    ]:
+        pooled, unpooled, other = get_output_from_standard_transformer_models(
+            encoder=encoder, input_ids=input_ids, segment_ids=segment_ids, input_mask=input_mask,
+        )
+    elif model_arch in [
+        ModelArchitectures.XLM,
+    ]:
+        pooled, unpooled, other = get_output_from_xlm_with_lang_handing(
+            encoder=encoder, input_ids=input_ids, input_mask=input_mask, batch=batch,
+        )
+    elif model_arch in [
+        ModelArchitectures.BART,
+        ModelArchitectures.MBART,
+    ]:
+        pooled, unpooled, other = get_output_from_bart_models(
+            encoder=encoder, input_ids=input_ids, input_mask=input_mask,
+        )
+    else:
+        raise KeyError(model_arch)
+
+    # Extend later with attention, hidden_acts, etc
+    if other:
+        return EncoderOutput(pooled=pooled, unpooled=unpooled)
+    else:
+        return EncoderOutput(pooled=pooled, unpooled=unpooled, other=other)
+
+
+def get_output_from_standard_transformer_models(encoder, input_ids, segment_ids, input_mask):
     output = encoder(input_ids=input_ids, token_type_ids=segment_ids, attention_mask=input_mask)
```

Comment:
following the above, batch could be passed into an argument here. Then the encoder is responsible for returning `pooled, unpooled, other`